skimage/transform/_geometric.py

```diff
@@ -970,8 +1047,64 @@ def inverse(self, coords):
         return out
 
 
+def _euler_rotation(axis, angle):
+    """Produce a single-axis Euler rotation matrix.
+
+    Parameters
+    ----------
+    axis : int in {0, 1, 2}
+        The axis of rotation.
+    angle : float
+        The angle of rotation in radians.
+
+    Returns
+    -------
+    Ri : array of float, shape (3, 3)
+        The rotation matrix along axis `axis`.
+    """
+    i = axis
+    s, c = np.sin(angle), np.cos(angle)
+    R2 = np.array([[          c, (-1)**(i+1) * s],
+                   [(-1)**i * s,               c]])
+    Ri = np.eye(3)
+    # We need the axes other than the rotation axis, in the right order:
+    # 0 -> (1, 2); 1 -> (0, 2); 2 -> (0, 1).
+    axes = sorted({0, 1, 2} - {axis})
+    # We then embed the 2-axis rotation matrix into the full matrix.
+    # (1, 2) -> R[1:3:1, 1:3:1] = R2, (0, 2) -> R[0:3:2, 0:3:2] = R2, etc.
+    sl = slice(axes[0], axes[1] + 1, axes[1] - axes[0])
```

Comment:
Give me more magic
```suggestion
    args = [None, None, None]
    args[-axis] = 1 + ( axis > 0)
    sl = slice(*args)
```
:smiling_imp:

Reply:
That does look correct, but is pretty obscure (thus the devil emoji, I suppose!).

sorted `axes` could also be created with a generator expression:
```Python
# Get the (sorted) axes of the plane in which rotation occurs
ax1, ax2 = (ax for ax in range(3) if ax != axis)
# We then embed the 2-axis rotation matrix into the full matrix. 
sl = slice(ax1, ax2 + 1, ax2 - ax1)
```

Reply:
Imma keep my code + comments, thanks both. :joy: